src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,64 @@
+import classNames from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const currentSex = searchParams.get('sex') || '';
+
+  function handleQueryChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const params = new URLSearchParams(searchParams);
+
+    if (event.target.value) {
+      params.set('query', event.target.value);
+    } else {
+      params.delete('query');
+    }
+
+    setSearchParams(params);
+  }
+
+  function getCenturies(century: string) {
+    return centuries.includes(century)
+      ? centuries.filter(c => c !== century)
+      : [...centuries, century];
+  }
+
+  const isActiveSex = (value: string | null) => {
+    return value === currentSex || (value === null && !currentSex);
+  };
+
+  const resetFilters = {
+    sex: null,
+    centuries: null,
+    query: null,
```

Comment:
The `resetFilters` object sets all filter parameters to `null`. Ensure that the `SearchLink` component correctly interprets `null` values to reset the filters effectively. If `null` does not clear the parameters, consider using empty strings or removing the parameters instead.